mosviz/plugins/cutout_tool/cutout_tool.py

```diff
@@ -18,18 +16,103 @@
+        * ``'cutout_y_size'`` - Cutout height (e.g., in arcsec).
+        * ``'cutout_pa'`` - Cutout angle (e.g., in degrees). This is only
+          use if user chooses to rotate the cutouts. Positive value
+          will result in a clockwise rotation.
+        * ``'spatial_pixel_scale'`` - Pixel scale (e.g., in arcsec/pix).
+
+    The following are no longer used, so they are now optional:
+        * ``'slit_pa'`` - Slit angle (e.g., in degrees).
+        * ``'slit_width'`` - Slit width (e.g., in arcsec).
+        * ``'slit_length'`` - Slit length (e.g., in arcsec).
+
+    Cutouts are organized as follows::
+        working_dir/
+            &lt;image_label&gt;_cutouts/
+                &lt;id&gt;_&lt;image_label&gt;_cutout.fits
+
+    Each cutout image is a simple single-extension FITS with updated WCS.
+    Its header has the following special keywords:
+        * ``OBJ_RA`` - RA of the cutout object in degrees.
+        * ``OBJ_DEC`` - DEC of the cutout object in degrees.
+        * ``OBJ_ROT`` - Rotation of cutout object in degrees.
+    Can add Qt.WindowStaysOnTopHint to supper to keep window ontop.
+
+    Parameters
+    ----------
+    table : QTable
+        Catalog table defining the sources to cut out.
+    imagename : str
+        Image to cut.
+    image_label : str
+        Label to name the cutout sub-directory and filenames.
+    output_file_format : str, optional
+        Typically '{0}.fits'
+    output_dir_format : str, optional
+        Typically '{0}_cutouts'
+    apply_rotation : bool, optional
+        Cutout will be rotated to a given angle. Default is `False`.
+    image_ext : int, optional
+        Image extension to extract header and data. Default is 0.
+    clobber : bool, optional
+        Overwrite existing files. Default is `False`.
+    verbose : bool, optional
+        Print extra info. Default is `True`.
+    report : function, optional
+        Call back that allows monitoring of the internal algorithm by the
+        caller. Set to None if no reporting is needed.
+    """
+    # Define output names.
+    if not output_file_format:
+        output_file_format = '{0}.fits'
+    if not output_dir_format:
+        output_dir_format = '{0}_cutouts'
+
+    # Sub-directory, relative to working directory.
+    path = output_dir_format.format(image_label)
+    if not os.path.exists(path):
+        os.mkdir(path)
+
+    table.rename_column("cutout_x_size", "cutout_width")
+    table.rename_column("cutout_y_size", "cutout_height")
+
+    fits_cutouts = cutouts_from_fits(imagename, table, output_dir=path,
+                                     overwrite=True, verbose=True,
+                                     report=report)
+    if ispreview:
+        return None
+    else:
+        success_list = [True if type(e)==fits.hdu.image.PrimaryHDU else False
+                        for e in fits_cutouts]
+        success_counter = success_list.count(True)
+        return success_counter, success_list
```

Comment:
1. The resulting `fits_cutouts` should be returned along with `success_counter, success_list`
2. There should not be a `is preview` check. The function should always return all 3 items.

Reply:
Done.